uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -1,30 +1,64 @@
 import asyncio
 import http
 import logging
+import sys
+from typing import Any, List, Optional, Sequence, Tuple, Union, cast
 from urllib.parse import unquote
 
 import websockets
+from trustme import TYPE_CHECKING
+from websockets.datastructures import Headers
+from websockets.exceptions import ConnectionClosed
 from websockets.extensions.permessage_deflate import ServerPerMessageDeflateFactory
+from websockets.legacy.server import HTTPResponse
+from websockets.server import WebSocketServerProtocol
+from websockets.typing import Subprotocol
 
+from uvicorn.config import Config
 from uvicorn.logging import TRACE_LOG_LEVEL
 from uvicorn.protocols.utils import get_local_addr, get_remote_addr, is_ssl
+from uvicorn.server import ServerState
+
+if sys.version_info < (3, 8):
+    from typing_extensions import Literal
+else:
+    from typing import Literal
+
+if TYPE_CHECKING:
+    from asgiref.typing import (
+        ASGISendEvent,
+        WebSocketAcceptEvent,
+        WebSocketCloseEvent,
+        WebSocketConnectEvent,
+        WebSocketDisconnectEvent,
+        WebSocketReceiveEvent,
+        WebSocketScope,
+        WebSocketSendEvent,
+    )
 
 
 class Server:
     closing = False
 
-    def register(self, ws):
+    def register(self, ws: WebSocketServerProtocol) -> None:
```

Comment:
I guess it's needed since our `WebSocketProtocol` inherits from `WebSocketServerProtocol` ?

Reply:
Exactly.